### PR TITLE
oie changes for registration hooks

### PR DESCRIPTION
**Pull request by colinwang-okta**
## Description:
- Update Registration Hooks Documentation for OIE

### Resolves:

* [OKTA-404956](https://oktainc.atlassian.net/browse/OKTA-404956)

### CODE REVIEW

**Review by brianduffield-okta — COMMENTED** [review]
Should we leave in both procedures and naming conventions, as, the Classic Engine will be supported for some time. @quyle-okta can you confirm?

**Comment by quyle-okta** [comment]
@brianduffield-okta Yes - we'll need to include both instructions for Classic and OIE since Classic will be supported for awhile until Orgs upgrade to OIE which will take awhile. Some Orgs also may not choose to upgrade to OIE until they find enough value to upgrade.

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 1]
```diff
@@ -233,7 +233,34 @@ If there is a response timeout after receiving the Okta request, the Okta proces
...
+
+> **Note:** This feature is only available as a part of the Okta Identity Engine. Please [contact support](mailto:dev-inquiries@okta.com) for further information.
+
+To activate the Inline Hook, you first need to register your external service endpoint with Okta; see [Inline Hook Setup](/docs/concepts/inline-hooks/#inline-hooks_setup).
+
+You then need to associate the registered Inline Hook with your Profile Enrollment policy. (For information on configuring a Profile Enrollment policy, see [Enable and configure a profile enrollment policy](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/policies/create-profile-enrollment-policy-sr.htm).)
+
+1. Go to **Security > Profile Enrollment**
+
+1. Click the pencil icon to edit the policy for which you want your registration hook to be associated with.
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:248]:
```suggestion
1. Click the pencil icon to edit the policy and associate it with your Registration Inline Hook.
```

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 2]
```diff
@@ -233,7 +233,34 @@ If there is a response timeout after receiving the Okta request, the Okta proces
...
+
+1. Select your hook from the dropdown menu for **Use the following inline hook** under the options for **For new users**. If you have created multiple Registration Inline Hooks, you should see all of them displayed here.
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:250]:
```suggestion
1. Select your hook from the drop-down menu for **Use the following inline hook** under the options for **For new users**. If you have created multiple Registration Inline Hooks, you should see all of them displayed here.
```

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 3]
```diff
@@ -233,7 +233,34 @@ If there is a response timeout after receiving the Okta request, the Okta proces
...
+> **Note:** This feature is only available as a part of the Okta Identity Engine. Please [contact support](mailto:dev-inquiries@okta.com) for further information.
+
+To activate the Inline Hook, you first need to register your external service endpoint with Okta; see [Inline Hook Setup](/docs/concepts/inline-hooks/#inline-hooks_setup).
+
+You then need to associate the registered Inline Hook with your Profile Enrollment policy. (For information on configuring a Profile Enrollment policy, see [Enable and configure a profile enrollment policy](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/policies/create-profile-enrollment-policy-sr.htm).)
+
+1. Go to **Security > Profile Enrollment**
+
+1. Click the pencil icon to edit the policy for which you want your registration hook to be associated with.
+
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:249]:
```suggestion
1. In **Enrollment Settings**, click the More Options icon and then select **Edit**. If not already selected, click **Allowed** in the **For new users** section.
```

**Review by brianduffield-okta — CHANGES_REQUESTED** [review]
Added a step in the OIE procedure for registration hook. You may want to double-check that. Also, in some areas of the reference, there were updates to "Profile Enrollment" from "Self-Service Registration". Just want to make sure this is okay for Okta Classic, i.e, (do we need to mention both terms for now?)

**Comment by colinwang-okta** [comment]
@brianduffield-okta Thanks for the feedback! For the question of the change from Self-Service Registration to Profile Enrollment, the feedback from PM was to have the general parts mention the OIE flow (Profile Enrollment) vs the Classic flow (Self-Service Registration) so that is the reason for those changes. There is still a section in the bottom for setup for the classic flow, so I think it should still be clear enough what to do in the classic flow, but if you think we should mention Self-Service Registration every time we mention Profile Enrollment we could also do that.

**Review by brianduffield-okta — APPROVED** [review]
Thanks for your comments, @colinwang-okta. LGTM!

File: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md [hunk 4]
```diff
@@ -233,7 +233,36 @@ If there is a response timeout after receiving the Okta request, the Okta proces
...
+> **Note:** This feature is only available as a part of the Okta Identity Engine. Please [contact support](mailto:dev-inquiries@okta.com) for further information.
+
+To activate the Inline Hook, you first need to register your external service endpoint with Okta; see [Inline Hook Setup](/docs/concepts/inline-hooks/#inline-hooks_setup).
+
+You then need to associate the registered Inline Hook with your Profile Enrollment policy. (For information on configuring a Profile Enrollment policy, see [Enable and configure a profile enrollment policy](https://help.okta.com/oie/en-us/Content/Topics/identity-engine/policies/create-profile-enrollment-policy-sr.htm).)
+
+1. Go to **Security > Profile Enrollment**
+
+1. Click the pencil icon to edit the policy and associate it with your Registration Inline Hook.
```
Comment (petermiller-okta) [inline packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:248]: Are these numbered items supposed to increment?

Comment (colinwang-okta) [reply packages/@okta/vuepress-site/docs/reference/registration-hook/index.md:248]: they become incremental when the markdown is rendered

**Review by petermiller-okta — APPROVED** [review]

**Pull request merged**